backend/server/repositories/bucket_repository.ts

```diff
@@ -1,11 +1,11 @@
-const Bucket = require('../models/bucket');
-const Promise = require('bluebird');
-const logger = require('../logger');
-const PageUtils = require('../utils/page_utils');
+import logger from '../logger';
+import { BucketModel } from '../models/bucket';
+import * as Promise from 'bluebird';
```

Comment:
I wonder if typescript already has the Promisey stuff needed to join 2 promises into one